src/vs/workbench/contrib/tasks/browser/terminalTaskSystem.ts

```diff
@@ -943,7 +943,16 @@ export class TerminalTaskSystem extends Disposable implements ITaskSystem {
 				});
 			});
 			if (trigger === Triggers.reconnect && !!terminal.xterm) {
-				const bufferLines = terminal.xterm.getBufferLines(watchingProblemMatcher.beginPatterns);
+				const bufferLines = [];
+
+				const bufferReverseIterator = terminal.xterm.getBufferReverseIterator();
+				const startRegex = new RegExp(watchingProblemMatcher.beginPatterns.join('|'));
```

Comment:
I think this will result in incorrect regex's so it will always read everything?

[RegExp, RegExp].join('|') gives '/first/|/second/' but you want `first|second`?